rules/DowngradePhp81/Rector/FuncCall/DowngradeHashAlgorithmXxHashRector.php

```diff
@@ -0,0 +1,194 @@
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Tests\DowngradePhp81\Rector\FuncCall\DowngradeHashAlgorithmXxHash\DowngradeHashAlgorithmXxHashRectorTest
+ */
+final class DowngradeHashAlgorithmXxHashRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    private const HASH_ALGORITHMS_TO_DOWNGRADE = [
+        'xxh32' => MHASH_XXH32,
+        'xxh64' => MHASH_XXH64,
+        'xxh3' => MHASH_XXH3,
+        'xxh128' => MHASH_XXH128,
+    ];
+
+    /**
+     * @var array<string, string>
+     */
+    private array $hashAlgorithmsToDowngradeMapping = [];
+
+    private int $argNamedKey;
+
+    public function __construct(
+        private readonly ArgsAnalyzer $argsAnalyzer,
+        private readonly ValueResolver $valueResolver,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Downgrade hash algorithm xxh32, xxh64, xxh3 or xxh128 by default to md5. You can configure the algorithm to downgrade.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return hash('xxh128', 'some-data-to-hash');
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return hash('md5', 'some-data-to-hash');
+    }
+}
+CODE_SAMPLE
+                ),
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return hash('xxh3', 'some-data-to-hash');
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return hash('sha1', 'some-data-to-hash');
+    }
+}
+CODE_SAMPLE
+                    ,
+                    [
+                        'xxh3' => 'sha1',
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $this->argNamedKey = 0;
+
+        $algorithm = $this->getHashAlgorithm($node->getArgs());
+
+        if (! array_key_exists($algorithm, self::HASH_ALGORITHMS_TO_DOWNGRADE)) {
+            return null;
+        }
+
+        $newAlgorithm = $this->hashAlgorithmsToDowngradeMapping[$algorithm] ?? 'md5';
+
+        $arg = $node->args[$this->argNamedKey];
+
+        if (! $arg instanceof Arg) {
```

Comment:
this can be checked by : 

```php
if ($node->isFirstClassCallable()) {
    return null;
}

Reply:
If I change to your suggestion. PHPStan output following error
```
 ---------------------------------------------------------------------------------------------------------
  rules/DowngradePhp81/Rector/FuncCall/DowngradeHashAlgorithmXxHashRector.php:135
 ---------------------------------------------------------------------------------------------------------
  - '#Access to an undefined property PhpParser\\Node\\Arg\|PhpParser\\Node\\VariadicPlaceholder\:\:\$value#'
 ---------------------------------------------------------------------------------------------------------
```

Reply:
You can add `@var Arg $arg` after that

Reply:
Another solution is call `->getArgs()` after that, so no need `@var`.

```php
if ($node->isFirstClassCallable()) {
    return null;
}

// this always get Arg data
$args = $node->getArgs();
$arg = $args[$this->argNamedKey];
```